pages/en/lb3/Polymorphic-relations.md

```diff
@@ -147,80 +282,77 @@ Then here's an example of a polymorphic hasManyThrough relation:
       "type": "hasMany",
       "model": "Picture",
-      "polymorphic": {
-        "foreignKey": "imageableId",
-        "discriminator": "imageableType"
-      },
       "through": "ImageLink",
-      "keyThrough": "authorId"
+      "polymorphic": "imageable"
     }
   },
 ...
 }
 ```
 
-Equivalently, in JavaScript:
+OPTIONAL in model Picture.
 
-{% include code-caption.html content="/common/models/Author.js" %}
-```javascript
-Author.hasMany(Picture, {
-  as: 'pictures',
-  polymorphic: {
-    foreignKey: 'imageableId',
-    discriminator: 'imageableType'
-  },
-  through: ImageLink,
-  keyThrough: 'authorId'
-});
-```
-
-## BelongsTo polymorphic relations
-
-Because you define the related model dynamically, you cannot declare it up front.
-So instead of passing in the related model (name), you specify the name of the polymorphic relation.
-
-{% include code-caption.html content="common/models/picture.json" %}
+{% include code-caption.html content="/common/models/Picture.json" %}
 ```javascript
+// Optional, define invert hasMany relation in Picture
 {
   "name": "Picture",
   "base": "PersistedModel",
   ...
   "relations": {
-    "author": {
-      "type": "belongsTo",
+    "authors": {
+      "type": "hasMany",
       "model": "Author",
-      "polymorphic": {
-        "foreignKey": "imageableId",
-        "discriminator": "imageableType"
-      }
+      "through": "ImageLink",
+      "invert": true,
+      "polymorphic": "imageable"
+    },
+    "readers": {
+      "type": "hasMany",
+      "model": "Reader",
+      "through": "ImageLink",
+      "invert": true,
+      "polymorphic": "imageable"
     }
   },
 ...
+}
 ```
 
-Or, in code:
+Equivalently, in JavaScript:
 
-{% include code-caption.html content="common/models/picture.js" %}
+{% include code-caption.html content="/server/boot/boot-script.js" %}
 ```javascript
-Picture.belongsTo('imageable', {
-  polymorphic: true
-}); 
-// Alternatively, use an object for setup
-Picture.belongsTo('imageable', {
+Author.hasMany(Picture, {
+  as: 'pictures',
   polymorphic: {
     foreignKey: 'imageableId',
     discriminator: 'imageableType'
-  }
+  },
+  through: ImageLink
+});
+Reader.hasMany(Picture, {
+  as: 'pictures',
+  polymorphic: {
+    foreignKey: 'imageableId',
+    discriminator: 'imageableType'
+  },
+  through: ImageLink
 });
+ImageLink.belongsTo(Picture, {});
+ImageLink.belongsTo(ImageLink, {polymorphic: true});
+
+// Optional define invert hasMany relation in Picture
+Picture.hasMany(Author, {through: ImageLink, polymorphic: 'imageable', invert: true});
+Picture.hasMany(Reader, {through: ImageLink, polymorphic: 'imageable', invert: true});
 ```
 
 ## HasAndBelongsToMany polymorphic relations
 
-This requires an explicit 'through' model, in this case: PictureLink
-
-The relations `Picture.belongsTo(PictureLink)` and `Picture.belongsTo('imageable', { polymorphic: true });` will be setup automatically.
+hasAndBelongsToMany polymorphic relation is similar to hasManyThrough relation.
```

Comment:
"...similar to hasManyThrough relation."
Should be:
"similar to \[hasManyThrough relation\]\(#HasManyThrough-polymorphic-relations\)."